### DISCUSSION TITLE
Concourse deployment of v7.0.0 with "502 Bad Gateway Error" and "http: TLS handshake error"

### DISCUSSION THREAD

**Discussion by TiaguPinto** [concourse/concourse#6623] [Help & Support]
Hello everyone,

I had a deployment running of Concourse v6.7.5, but when I upgraded it to latest version v7.0.0, I am facing "**502 Bad Gateway** error on the browser, and when looking for the logs on the web node, I can see on **web.stderr.log**  :  **http: TLS handshake error from 10.247.0.54:42498: tls: client didn't provide a certificate**

**Triaging info**

**Concourse version:** 7.0.0
**Did this used to work?** Yes on 6.7.4 and 6.7.5 
**Browser**: Safari and Chrome 
**Runtime**: BOSH

**To deploy concourse on BOSH I am using the following command :** 
```
  echo "> Deploying concourse"
  bosh -e "$FOUNDATION" deploy --non-interactive \
    -d concourse "${CONCOURSE_REPOSITORY}/cluster/concourse.yml" \
    -l "${CONCOURSE_REPOSITORY}/versions.yml" \
    --vars-store "${CONCOURSE_REPOSITORY}/cluster/cluster-creds.yml" \
    -o "${CONCOURSE_REPOSITORY}/cluster/operations/basic-auth.yml" \
    -o "${CUSTOM_BOSH_FILES}/bind-ports.yml" \
    -o "${CONCOURSE_REPOSITORY}/cluster/operations/tls-vars.yml" \
    -o "${CONCOURSE_REPOSITORY}/cluster/operations/tls.yml" \
    --var local_user.username=admin \
    --var local_user.password="$CONCOURSE_PASSWORD" \
    --var web_ip="$STATIC_CONCOURSE_IP_MAIN" \
    --var external_url="$EXTERNAL_URL" \
    --var external_host="$EXTERNAL_URL" \
    --var network_name=concourse \
    --var web_vm_type=web_instance \
    --var db_vm_type=concourse \
    --var db_network_name=db \
    --var db_persistent_disk_type=db \
    --var db_az=[z3] \
    --var worker_vm_type=concourse \
    --var worker_network_name=worker \
    --var deployment_name=concourse \
    --var azs=[z1]
```
Anyone faced similar issue ? Any idea what could be leading to this issue ? 

If I make a deploy with 6.7.4 or 6.7.5 it works just fine.

Thanks in advance, Regards
Tiago Pinto

**Comment by tahiriah** [comment]
Hi,

I have the same issue after migrating to 7.0 from version 6.7.

Any update on this issue?


Best regards,
Ahmed

**Comment by taylorsilva** [comment]
Where are you seeing `502 Bad Gateway` on the web UI? On the home screen? Some build pages? Only on the login screen?

I've seen `http: TLS handshake error from 10.247.0.54:42498: tls: client didn't provide a certificate` in `web.stderr.log` many times before and I think it's an error that happens during startup and eventually resolves itself. I would ignore it. I would look deeper in the stdout logs instead for any logs with level `error`.

**Comment by alexnguyen91** [comment]
`    -o "${CONCOURSE_REPOSITORY}/cluster/operations/tls-vars.yml" \
    -o "${CONCOURSE_REPOSITORY}/cluster/operations/tls.yml" \`
Using those options will set `CONCOURSE_TLS_CA_CERT` which will enable mTLS due to PR https://github.com/concourse/concourse/pull/6355/files

**Reply by o-orand** [reply]
using latest version of concourse-bosh-deployment does not generate this issue anymore, see https://github.com/concourse/concourse-bosh-deployment/pull/234
